server/src/domain/shared-link/shared-link.service.ts

```diff
@@ -119,15 +119,19 @@ export class SharedLinkService {
       throw new BadRequestException('Invalid shared link type');
     }
 
+    const existingAssetIds = new Set(sharedLink.assets.map((asset) => asset.id));
+    const notPresentAssetIds = dto.assetIds.filter((assetId) => !existingAssetIds.has(assetId));
+    const allowedAssetIds = await this.access.checkAccess(authUser, Permission.ASSET_SHARE, notPresentAssetIds);
+
     const results: AssetIdsResponseDto[] = [];
     for (const assetId of dto.assetIds) {
-      const hasAsset = sharedLink.assets.find((asset) => asset.id === assetId);
+      const hasAsset = existingAssetIds.has(assetId);
```

Comment:
We should change this to not require loading all the assets into memory for this to work, but I think this is the best we can do in the meantime.

Reply:
For this an other places where we get `existingAssetIds` in memory, I think we can have a separate PR where we chunk the ids to limit the amount of them in memory. There are currently three places where the same pattern could be improved, so I'd prefer to maintain it separate from this change.

Reply:
Yes, agreed